src/NzbDrone.Common/Disk/DiskProviderBase.cs

```diff
@@ -360,6 +361,31 @@ public void SetPermissions(string filename, WellKnownSidType accountSid, FileSys
 
         }
 
+        protected static string GetFolderPermissions(string filePermissions)
+        {
+            var builder = new StringBuilder();
```

Comment:
This works. But I think it's cleaner to manipulate the output of `FilePermissions fp = NativeConvert.FromOctalPermissionString(mask)` that we're already getting.
you can even just mask out the 3 'r' (see FilePermissions), shift them down 2 bits and or.

Reply:
Sure, I'll change it.

Reply:
forgot to mention that you also need to make sure the user doesn't add executable flag to filechmod. Either through validation, or masking them out later.

Reply:
I'll improve the validation then, because currently it only checks if it's not empty.